### ISSUE TITLE
[Bug] Crash on returning from a new private tab

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#8580]
## Steps to reproduce
1. Open a tab in normal browsing mode.
2. From the main menu, tap Private tab.
3. The private search is opened, tap the back button to dismiss it.
### Expected behavior
Private mode is dismissed and the normal browsing tab is displayed.
### Actual behavior
App crashes. 

[crash log.txt](https://github.com/mozilla-mobile/fenix/files/4231172/crash.log.txt)

* Android device: On some devices, it crashes more often than on others: e.g Samsung Galaxy S8 (Android 9) 1/2 times, Xiaomi Mi Pad 2 (Android 5.1) every time.
### Device information
* Fenix version: Beta 4.0 - build1.

**Comment by AaronMT** [comment]
```Kotlin
2020-02-20 10:59:19.860 13087-13087/org.mozilla.fenix.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix.debug, PID: 13087
    java.lang.IllegalStateException: FragmentManager has been destroyed
        at androidx.fragment.app.FragmentManager.ensureExecReady(FragmentManager.java:1783)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1843)
        at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
        at android.os.Handler.handleCallback(Handler.java:883)
```

**Comment by ekager** [comment]
Similar to #8374
Sentry crash recorded in #8413

**Comment by ekager** [comment]
This also happens the opposite direction - ie from private mode, open a new tab, press back

**Comment by boek** [comment]
@ekager I think this might have something to do with the browsermode refactor

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on the latest `Nightly Build #20550606 from 02/24` using the following devices:
• Google Pixel 3a XL (Android 10)
• Google Pixel 3a (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7)
• OnePlus A3 (Android 6.0.1)
• LG Nexus 4 (Android 5.1.1)

► Video
![20200224-090229](https://user-images.githubusercontent.com/51314259/75133923-38488e00-56e5-11ea-9ed9-2e3c8ed80bf4.gif)
